### PR TITLE
Add etcd auth support

### LINKED ISSUE #1713: request help: add auth for etcd
https://github.com/api7/lua-resty-etcd/commit/7f95abff051308a75d251a98c4a50f8b095c0175 support etcd auth, do we need upgrade to etcd v3 first? @nic-chen

### LINKED ISSUE #1767: feature: support etcd v3
related issues:

https://github.com/apache/incubator-apisix/issues?q=is%3Aissue+is%3Aopen+v3+

`lua-resty-etcd` already supports etcd v3, we can support it by this feature:

https://github.com/api7/lua-resty-etcd/blob/master/api_v3.md

### PR DESCRIPTION
**Pull request by shenal**
### Summary 
This PR adds etcd auth support 
### Full changelog

* Added etcd auth placeholder configs
* Added etcd tests with auth pass & auth fail

### Issues resolved

Fix #1713  , #1770

### CODE REVIEW

**Comment by membphis** [comment]
nice PR.

File: t/core/etcd-auth-fail.t [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+use t::APISIX 'no_plan';
+
+repeat_each(1);
+no_long_string();
+no_root_location();
+log_level("info");
+
+# Authentication is enabled at etcd and credentials are set
+system('etcdctl --endpoints="http://127.0.0.1:2376" -u root:5tHkHhYkjr6cQY user add root:5tHkHhYkjr6cQY');
```
Comment (moonming) [inline t/core/etcd-auth-fail.t:29]: the APISIX node need install `etcdctl` first and run this command, then can enable etcd auth?
How about call etcd auth method in Lua land?

Comment (shenal) [reply t/core/etcd-auth-fail.t:29]: Thanks for the pointer @moonming . When I tested this code in my local etcdctl was installed along with etcd. I will check your suggestion

**Comment by shenal** [comment]
@membphis , @moonming Current travis runners use xenial(16.04) are you ok if I use bionic(18.04)

File: t/core/etcd-auth-fail.t (hunk 1 above)
Comment (membphis) [reply t/core/etcd-auth-fail.t]: > How about call etcd auth method in Lua land?

`lua-resty-etcd` does not support the manager API now.

**Comment by membphis** [comment]
> @membphis , @moonming Current Travis runners use xenial(16.04) are you ok if I use bionic(18.04)

we can do this in a new PR. one PR does one thing.

File: t/core/etcd-auth-fail.t (hunk 1 above)
Comment (shenal) [reply t/core/etcd-auth-fail.t]: This issue has been resolved. Tests are passing now

File: t/core/etcd.t [hunk 2]
```diff
@@ -0,0 +1,59 @@
+#
```
Comment (membphis) [inline t/core/etcd.t:1]: `t/core/etcd.t` -> `t/core/etcd-auth.t`

Comment (shenal) [reply t/core/etcd.t]: Done

**Comment by membphis** [comment]
@shenal merged, many thx

**Pull request merged**